src/Symfony/Component/PropertyAccess/StringUtil.php

```diff
@@ -203,7 +203,7 @@ public static function singularify($plural)
         }
 
         // Convert teeth to tooth, feet to foot
-        if (false !== ($pos = strpos($plural, 'ee')) && strlen($plural) > 3 && 'feedback' !== $plural) {
+        if (false !== ($pos = strpos($plural, 'ee')) && strlen($plural) > 3 && !preg_match('{feedback|Feedback|career|Career}', $plural)) {
```

Comment:
Hi Nicolas.
That is intended. Please, notice that in my example, the false plural is a composed string: "careerSubject", so anchoring the regex or ```in_array()``` aren't an option here.